sdk/textanalytics/ai-text-analytics/src/analyzeBatchActionsResult.ts

```diff
@@ -47,3 +155,308 @@ export interface PagedAnalyzeBatchActionsResult
+    const { results: actionResults, lastUpdateDateTime } = task;
+    if (actionResults.documents.length !== 0 || actionResults.errors.length !== 0) {
+      const recognizeEntitiesResults = makeRecognizeCategorizedEntitiesResultArray(
+        documents,
+        actionResults?.documents,
+        actionResults?.errors,
+        actionResults?.modelVersion,
+        actionResults?.statistics
+      );
+      return [
+        ...actions,
+        {
+          results: recognizeEntitiesResults,
+          completedOn: lastUpdateDateTime
+        }
+      ];
+    } else {
+      return actions;
+    }
+  }
+
+  const succeededActions = succeededTasks.reduce(convertTasksToActions, []);
+  return combineSucceededAndErredActions(succeededActions, erredActions);
+}
+
+/**
+ * Creates a list of results for recognize pii entities actions.
+ * @param documents - list of input documents
+ * @param succeededTasks - list of succeeded action results
+ * @param erredActions - list of erred actions
+ * @internal
+ */
+function makeRecognizePiiEntitiesActionResult(
+  documents: TextDocumentInput[],
+  succeededTasks: TasksStateTasksEntityRecognitionPiiTasksItem[],
+  erredActions: TextAnalyticsActionError[]
+): RecognizePiiEntitiesActionResult[] {
+  function convertTasksToActions(
+    actions: RecognizePiiEntitiesActionSuccessResult[],
+    task: TasksStateTasksEntityRecognitionPiiTasksItem
+  ): RecognizePiiEntitiesActionSuccessResult[] {
+    const { results: actionResults, lastUpdateDateTime } = task;
+    if (actionResults.documents.length !== 0 || actionResults.errors.length !== 0) {
+      const recognizeEntitiesResults = makeRecognizePiiEntitiesResultArray(
+        documents,
+        actionResults
+      );
+      return [
+        ...actions,
+        {
+          results: recognizeEntitiesResults,
+          completedOn: lastUpdateDateTime
+        }
+      ];
+    } else {
+      return actions;
+    }
+  }
+
+  const succeededActions = succeededTasks.reduce(convertTasksToActions, []);
+  return combineSucceededAndErredActions(succeededActions, erredActions);
+}
+
+/**
+ * Creates a list of results for extract key phrases actions.
+ * @param documents - list of input documents
+ * @param succeededTasks - list of succeeded action results
+ * @param erredActions - list of erred actions
+ * @internal
+ */
+function makeExtractKeyPhrasesActionResult(
+  documents: TextDocumentInput[],
+  succeededTasks: TasksStateTasksKeyPhraseExtractionTasksItem[],
+  erredActions: TextAnalyticsActionError[]
+): ExtractKeyPhrasesActionResult[] {
+  function convertTasksToActions(
+    actions: ExtractKeyPhrasesActionSuccessResult[],
+    task: TasksStateTasksKeyPhraseExtractionTasksItem
+  ): ExtractKeyPhrasesActionSuccessResult[] {
+    const { results: actionResults, lastUpdateDateTime } = task;
+    if (actionResults.documents.length !== 0 || actionResults.errors.length !== 0) {
+      const recognizeEntitiesResults = makeExtractKeyPhrasesResultArray(
```

Comment:
kind of confused since it says it's a `recognizeEntitiesResults` but it's actually an extract key phrases result. I think it's just a local variable rename though

Reply:
Yeah that was copied, I fixed it.